alphagov/paas-cf
# [#154769249] Match latest cf deployment versions

## What

Upgrade our bosh releases to match those of cf-deployment v1.14.0, so that we're closer to cf-deployment and can more easily transition to it.

We introduced a `legacy` stemcell version to keep our ancient logsearch deployment happy. See https://github.com/cloudfoundry-community/logsearch-boshrelease/issues/115 for details.

We have split the buildpacks out to #1238 because we need to give tenants a week notice before the upgrade and don't want to hold the entire upgrade that long.

The following was noted upstream but should happen automatically for us because we are also changing stemcells:

> This release of cf-deployment includes an update of garden-runc-release that uses GrootFS by default. As such, there will be changes to some ops-files (see below). **An important note for operators**: it is recommended to recreate cells shortly after upgrading; we've managed to include a stemcell update as part of this release, so that should happen automatically for anybody using the stemcell specified in cf-deployment.yml.

## How to review

1. Deploy to a development environment that is already running from `master`
     - although we've tested this it's definitely worth repeating from a clean state because it's very hard to undo changes to our already upgraded dev envs.
1. Check that all of the tests in the pipeline pass.
1. Check versions match those of [cf-deployment 1.14.0](https://github.com/cloudfoundry/cf-deployment/releases/tag/v1.14.0).
1. 🐝-fore merging: review and merge alphagov/paas-docker-cloudfoundry-tools#117, then update the `wip` commit to use the new tag.

## Who can review

Not @camelpunch or @LeePorte or @dcarley
